docs/SRIOV.md

```diff
@@ -0,0 +1,260 @@
+firmware-version: 2.42.5000
+expansion-rom-version:
+bus-info: 0000:08:00.0
+supports-statistics: yes
+supports-test: yes
+supports-eeprom-access: no
+supports-register-dump: no
+supports-priv-flags: yes
+```
+
+```
+root@compute02:~# ethtool -i ens1f0
+driver: ixgbe
+version: 5.1.0-k
+firmware-version: 0x61bd0001
+expansion-rom-version:
+bus-info: 0000:08:00.0
+supports-statistics: yes
+supports-test: yes
+supports-eeprom-access: yes
+supports-register-dump: yes
+supports-priv-flags: yes
+```
+
+The **host_vars** for each host can be implemented in a file that corresponds to the host's short name located at **/opt/pf9-express/host_vars/<shortname>.yml**. In the following example, **compute01** uses a single network interface for SR-IOV, while **compute02** uses two. SR-IOV networks will leverage a new provider label named **sriov** and 8 VFs per interface, as shown here:
+
+```
+---
+# compute01.yml
+physical_device_mappings:
+  - sriov:ens1
+sriov_numvfs:
+  - ens1:8
+```
+
+```
+---
+# compute02.yml
+physical_device_mappings:
+  - sriov:ens1f0
+  - sriov:ens1f1
+sriov_numvfs:
+  - ens1f0:8
+  - ens1f1:8
+```
+
+> SR-IOV supports VLAN networks only. Flat and overlay networks are not supported.
+
+### Group Variables
+
+Group-wide configurations can be implemented using what is known as **group_vars**. Configurations that may be consistent between groups include:
+
+* Network interface name
+* Number of VFs per interface
+* Provider network mappings
+
+Using **group_vars**, the following are some variables that can be modified:
+
+* neutron_ovs_bridge_mappings
+* sriov_numvfs
+* physical_device_mappings
+
+The **group_vars** for the **hypervisors** group can be implemented in a file that corresponds to the group's name located at **/opt/pf9-express/group_vars/<groupname>.yml**. In the following example, every host in the **hypervisors** group has the same NIC installed in the same slot, so the naming convention is consistent across all hosts. A second provider bridge mapping has been established that will allow non-SR-IOV capable ports, such as DHCP, to connect to a vSwitch and communicate with SR-IOV ports:
+
+```
+---
+# hypervisors.yml
+...
+neutron_ovs_bridge_mappings: "external:br-pf9, sriov:br-sriov"
+physical_device_mappings:
+  - sriov:ens1f0
+  - sriov:ens1f1
+sriov_numvfs:
+  - ens1f0:8
+  - ens1f1:8
+...
+```
+
+> Host vars take precedence over group vars. If a small number of hosts vary from the greater group, feel free to implement the respective **host_vars** files accordingly.
+
+### Inventory File
+
+To enable support for SR-IOV on a host, the inventory must be modified according so that SR-IOV related tasks are executed. One method of enabling support for a host is to add the **sriov=on** variable to an individual host in the **hypervisors** group, as shown here:
+
+```
+[hypervisors]
+compute01 ansible_host=10.50.0.197 vm_console_ip=10.50.0.197 ha_cluster_ip=10.50.0.197 tunnel_ip=10.50.0.197 dhcp=on snat=on sriov=on
```

Comment:
Rename to hv01 to align with readme / sample inventory file. Also would be good for sample IPs to align.

Reply:
fixed.